### PR TITLE
Remove Prometheus dependency from the Swarm chart

### PR DESCRIPTION
**Pull request by afalhambra-hivemq**
https://hivemq.kanbanize.com/ctrl_board/22/cards/22656/details/

### CODE REVIEW

**Review by mhofsche — COMMENTED** [review]
We probably also have to remove the charts.lock file or? 
Also can we add a hivemq-swarm folder in the examples folder and add the example grafana dashboard and an example servicemonitor.yaml?

**Comment by afalhambra-hivemq** [comment]
> We probably also have to remove the charts.lock file or?
For some reason, it didn't go to the commit I pushed. Thanks for noticing. Will address that.

> Also can we add a hivemq-swarm folder in the examples folder and add the example grafana dashboard and an example servicemonitor.yaml?
Good one, will do.

**Comment by afalhambra-hivemq** [comment]
@mhofsche it should be now ready for review again. It can also be reviewed along with the doc PR: https://github.com/hivemq/hivemq4-documentation/pull/1716

**Review by Donnerbart — APPROVED** [review]
I like this very much. It's better aligned with the principles of the Platform Helm charts and the additional effort for customers is very little, in comparison to the compatibility issues with K8s and our fixed Prometheus subchart versions.

File: examples/README.md [hunk 1]
```diff
@@ -8,7 +8,11 @@ Examples on how to deploy HiveMQ Platform clusters with the HiveMQ Platform Oper
 
 ### [hivemq-operator](./hivemq-operator)
 
-Examples for the legacy HiveMQ Operator
+Examples for the legacy HiveMQ Operator chart
+
+### [hivemq-swarm](./hivemq-swarm)
+
+Includes an example about how to use and install a Prometheus ServiceMonitor along with a custom Grafana dashboard through a ConfigMap for the HiveMQ Swarm chart
```
Comment (mhofsche) [inline examples/README.md:15]:
```suggestion
Examples for the HiveMQ Swam chart, including examples for Prometheus and Grafana monitoring.
```

File: examples/README.md [hunk 2]
```diff
@@ -8,7 +8,11 @@ Examples on how to deploy HiveMQ Platform clusters with the HiveMQ Platform Oper
...
 
 ## Documentation
 See the [HiveMQ Kubernetes documentation](https://docs.hivemq.com/hivemq-platform-operator/introduction.html) for more detailed information.
```
Comment (mhofsche) [inline examples/README.md:18]: Lets remove this last documentation section and instead add a documentation link to each section above.

File: examples/hivemq-swarm/configmap_monitoring.yaml
Comment (mhofsche) [inline examples/hivemq-swarm/configmap_monitoring.yaml:1]: would rename this file to `configmap_grafana_dashboard.yaml`

File: examples/hivemq-swarm/configmap_monitoring.yaml [hunk 3]
```diff
@@ -0,0 +1,2372 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: swarm-hivemq-swarm-dashboard
+  namespace: monitoring # Namespace where the prometheus-community/kube-prometheus-stack is installed
+  labels:
+    app.kubernetes.io/instance: swarm # Helm release-name of HiveMQ Swarm chart
+    app.kubernetes.io/name: hivemq-swarm
+    app.kubernetes.io/version: 4.32.0
```
Comment (mhofsche) [inline examples/hivemq-swarm/configmap_monitoring.yaml:9]: do we need this version information here? This outdates quickly

File: examples/hivemq-swarm/configmap_monitoring.yaml [hunk 4]
```diff
@@ -0,0 +1,2372 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: swarm-hivemq-swarm-dashboard
+  namespace: monitoring # Namespace where the prometheus-community/kube-prometheus-stack is installed
+  labels:
+    app.kubernetes.io/instance: swarm # Helm release-name of HiveMQ Swarm chart
```
Comment (mhofsche) [inline examples/hivemq-swarm/configmap_monitoring.yaml:7]:
```suggestion
    app.kubernetes.io/instance: swarm # CUSTOMIZE: Helm release-name of HiveMQ Swarm chart
```

File: examples/hivemq-swarm/configmap_monitoring.yaml [hunk 5]
```diff
@@ -0,0 +1,2372 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: swarm-hivemq-swarm-dashboard
+  namespace: monitoring # Namespace where the prometheus-community/kube-prometheus-stack is installed
```
Comment (mhofsche) [inline examples/hivemq-swarm/configmap_monitoring.yaml:5]:
```suggestion
  namespace: monitoring # CUSTOMIZE: Namespace where the prometheus-community/kube-prometheus-stack is installed
```

File: examples/hivemq-swarm/configmap_monitoring.yaml [hunk 6]
```diff
@@ -0,0 +1,2372 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: swarm-hivemq-swarm-dashboard
```
Comment (mhofsche) [inline examples/hivemq-swarm/configmap_monitoring.yaml:4]: Does this name also have to be customized? 
If possible I would change this to `hivemq-swarm-grafana-dashboard`

File: examples/hivemq-swarm/servicemonitor.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: hivemq-swarm-service-monitor
+  labels:
+    # Helm release-name of prometheus-stack
```
Comment (mhofsche) [inline examples/hivemq-swarm/servicemonitor.yaml:6]:
```suggestion
    # CUSTOMIZE: Helm release-name of prometheus-stack
```

File: examples/hivemq-swarm/servicemonitor.yaml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    release: monitoring-stack
+spec:
+  selector:
+    matchLabels:
+      # agent and commander pods labels
+      app.kubernetes.io/instance: swarm # Helm release-name of HiveMQ Swarm chart
```
Comment (mhofsche) [inline examples/hivemq-swarm/servicemonitor.yaml:12]:
```suggestion
      app.kubernetes.io/instance: swarm # CUSTOMIZE: Helm release-name of HiveMQ Swarm chart
```

File: examples/hivemq-swarm/servicemonitor.yaml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+      app.kubernetes.io/name: hivemq-swarm
+  namespaceSelector:
+    matchNames:
+      # namespace of the Swarm deployment
```
Comment (mhofsche) [inline examples/hivemq-swarm/servicemonitor.yaml:16]:
```suggestion
      # CUSTOMIZE: namespace of the Swarm deployment
```

File: examples/hivemq-swarm/configmap_monitoring.yaml (hunk 4 above)
Comment (afalhambra-hivemq) [reply examples/hivemq-swarm/configmap_monitoring.yaml:7]: These extra labels are not needed. Will remove them. Only the `grafana_dashboard` is needed.

File: examples/hivemq-swarm/configmap_monitoring.yaml (hunk 3 above)
Comment (afalhambra-hivemq) [reply examples/hivemq-swarm/configmap_monitoring.yaml:9]: These extra labels are not needed. Will remove them. Only the grafana_dashboard is needed.

**Review by mhofsche — APPROVED** [review]

**Pull request merged**